File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    static int maxsum;

    public int maxSubArray(int[] arr) {
        maxsum = arr[0];

        int currsum = 0;

        int n = arr.length;
        for (int i = 0; i < n; i++) {
            currsum = currsum + arr[i];

            maxsum = Math.max(maxsum, currsum);

            if (currsum < 0) {
                currsum = 0;
            }
        }
        return maxsum;

    }
}

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    if (!nums || nums.length === 0) {
        return 0;
    }
    let max_so_far = nums[0];
    let max_ending_here = nums[0];

    for (let i = 1; i < nums.length; i++) {
        max_ending_here = Math.max(nums[i], max_ending_here + nums[i]);
        max_so_far = Math.max(max_so_far, max_ending_here);
    }

    return max_so_far;
};